nexus/reconfigurator/execution/src/omicron_zones.rs

```diff
@@ -178,6 +181,42 @@ pub(crate) async fn clean_up_expunged_zones<R: CleanupResolver>(
     }
 }
 
+async fn oximeter_cleanup(
+    opctx: &OpContext,
+    datastore: &DataStore,
+    zone_id: OmicronZoneUuid,
+    log: &Logger,
+) -> anyhow::Result<()> {
+    // Record that this Oximeter instance is gone.
+    datastore
+        .oximeter_expunge(opctx, zone_id.into_untyped_uuid())
+        .await
+        .context("failed to mark Oximeter instance deleted")?;
+
+    // Reassign any producers it was collecting to other Oximeter instances.
+    match datastore
+        .oximeter_reassign_all_producers(opctx, zone_id.into_untyped_uuid())
+        .await
+        .context("failed to reassign metric producers")?
+    {
+        CollectorReassignment::Complete(n) => {
```

Comment:
Just checking, but there's no way for `n` to be less than "all the producers", right?

Reply:
Correct.